src/cl/base.cl

```diff
@@ -138,12 +138,17 @@ typedef global const double2* BigTab;
 
 #define KERNEL(x) kernel __attribute__((reqd_work_group_size(x, 1, 1))) void
 
+// Prototypes
+void read(u32 WG, u32 N, T2 *u, const global T2 *in, u32 base);
```

Comment:
Interesting, are these prototypes needed? does the compiler produce some warnings without them?

Reply:
Yes, they fix warnings. Presumably, the OpenCL compiler on MacOS is unnecessarily strict.

```
./build-release/prpll -ll 82589933
20240913 23:12:47  PRPLL 0.15-4-g3a1361d-dirty starting
20240913 23:12:47  config: -ll 82589933
20240913 23:12:47  device 0, OpenCL 1.1, unique id ''
20240913 23:12:47 82589933 config:
20240913 23:12:47 82589933 FFT: 4.50M 1K:9:256:1:0 (17.50 bpw)
20240913 23:12:47 82589933 In file included from <program source>:1:
In file included from transpose.cl:4:
base.cl:147:6: warning: no previous prototype for function 'read'
void read(u32 WG, u32 N, T2 *u, const global T2 *in, u32 base) {
     ^

20240913 23:12:47 82589933 In file included from <program source>:1:
In file included from transpose.cl:4:
base.cl:147:6: warning: no previous prototype for function 'read'
void read(u32 WG, u32 N, T2 *u, const global T2 *in, u32 base) {
     ^

20240913 23:12:47  Exception gpu_error: INVALID_WORK_GROUP_SIZE (-54) transpIn at src/clwrap.cpp:312 run
20240913 23:12:47  Bye
```